test/plugin/newrelic-fluentd-output_test.rb

```diff
@@ -46,6 +48,102 @@ class Fluent::Plugin::NewrelicOutputTest < Test::Unit::TestCase
 
       assert_equal(true, driver.instance.multi_workers_ready?)
     end
+
+    class MockChunk
+      def initialize(times, records)
+        @times = times
+        @records = records
+      end
+
+      def msgpack_each(&block)
+        @times.length.times do |i|
+          block.call(@times[i], @records[i])
+        end
+      end
+    end
+
+    def create_driver_with_mocked_send_payload
+      driver = Fluent::Plugin::NewrelicOutput.new()
+
+      def driver.send_payload(payload)
+        @send_payload_called_with ||= []
+        @send_payload_called_with << payload
+      end
+
+      driver
+    end
+
+    sub_test_case "write()" do
+      test "one log line" do
+        driver = create_driver_with_mocked_send_payload
+        chunk = MockChunk.new([123],[{ "hello" => "world"}])
+
+        driver.write(chunk)
+
+        actual = driver.instance_variable_get(:@send_payload_called_with)
+        assert(actual.length == 1)
+        # There is something non-deterministic about the packaging of the
+        # message. We can't test for exact length or content
+        assert(actual[0].length > 100)
+      end
+
+      def big_strings
+        @big_strings ||= 20.times.map { |_| rand(36**100000).to_s(36) }
+      end
+
+      test "one log line that passes the max size" do
+        driver = create_driver_with_mocked_send_payload
+        chunk = MockChunk.new([123],[{ "hello" => big_strings.join }])
+
+        driver.write(chunk)
+
+        actual = driver.instance_variable_get(:@send_payload_called_with)
+        assert(actual.nil?)
+      end
+
+      test "two log lines split into two payloads" do
+        # 2 log lines, together too big, but separately not
+        driver = create_driver_with_mocked_send_payload
+        data = [
+          {"first" => big_strings[0, 10]},
+          {"second" => big_strings[10, 20]}
+        ]
+        chunk = MockChunk.new([123, 124], data)
+
+        driver.write(chunk)
+
+        actual = driver.instance_variable_get(:@send_payload_called_with)
+        assert(actual.length == 2)
+      end
+
+      test "5 log lines, one dropped" do
+        # 5 log lines, one is huge
+        driver = create_driver_with_mocked_send_payload
+        data = [
+          {"first" => "hello"},
+          {"second" => big_strings.join },
```

Comment:
How do you know, from the assessment at the end of the test, that this record was discarded? Is there a way to assess at least that you got 4 compressed log records back, instead of just 5?

Reply:
Done!

Reply:
Thanks!